sql/rpl_rli_pdb.cc

```diff
@@ -830,6 +830,102 @@ static void move_temp_tables_to_entry(THD *thd, db_worker_hash_entry *entry) {
   }
 }
 
+static double calculate_imbalance(Relay_log_info *rli,
+                                  const Slave_worker_array *ws) {
+  DBUG_ENTER("calculate_imbalance");
+
+  mysql_mutex_assert_owner(&rli->slave_worker_hash_lock);
+
+  // case: if the number of workers == number of db's
+  // nothing can be done to rebalance the workers
+  if (ws->size() == rli->mapping_db_to_worker.size()) DBUG_RETURN(0);
+
+  double max_usage = 0;
+  double total_usage = 0;
+  std::vector<ulong> load_on_workers(ws->size());
```

Comment:
Using/typedef for a custom container:
```suggestion
  using load_container = std::vector<ulong>;
  load_container load_on_workers(ws->size());
```

Reply:
I think this should be a softer rule in case of simple containers and no violations of DRY, like here